src/extra_context.rs

```diff
@@ -0,0 +1,42 @@
+//! ExtraContext is used for passing extra data to Vector's components when Vector is used as a library.
+use std::{
+    marker::{Send, Sync},
+    sync::Arc,
+};
+
+use anymap::{
+    any::{Any, IntoBox},
+    Map,
+};
+
+/// Structure containing any extra data.
+/// The data is held in an [`Arc`] so is cheap to clone.
+#[derive(Clone)]
+pub struct ExtraContext {
+    context: Arc<Map<dyn Any + Send + Sync>>,
+}
```

Comment:
> I'm also not excited about the type (shared any map) being named for its usage point (extra app context), but this is all more bikeshedding than anything.

I'm not quite sure what you mean.. Can you expound?

Reply:
The type is named `ExtraContext`. The whole name of the type is an indication of where it is used rather than what it contains. That is, this shared any-map is used as a context field that is extraneous to other application data. The type itself is not limited to being "extra context" and that term describes little about what it actually is and what range of operations can be done on it.

Anyways, like I said, more bikeshedding than anything on this point.